pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -297,20 +369,54 @@ func (d *DeploymentController) RenderServiceAccountApply(input MergedInput) *cor
 	// 	WithAPIVersion(gvk.KubernetesGateway.GroupVersion()))
 }
 
-// ApplyTemplate renders a template with the given input and (server-side) applies the results to the cluster.
-func (d *DeploymentController) ApplyTemplate(template string, input metav1.Object, subresources ...string) error {
-	var buf bytes.Buffer
-	if err := d.templates.ExecuteTemplate(&buf, template, input); err != nil {
-		return err
+type derivedInput struct {
+	TemplateInput
+
+	// Inserted from injection config
+	ProxyImage  string
+	ProxyConfig *meshapi.ProxyConfig
+	MeshConfig  *meshapi.MeshConfig
+	Values      map[string]any
+}
+
+func (d *DeploymentController) render(templateName string, mi TemplateInput) ([]string, error) {
+	cfg := d.injectConfig()
+
+	// TODO watch for template changes, update the Deployment if it does
```

Comment:
I believe this should have been resolved as a part of my PR.  Sorry for hanging reference.